### ISSUE TITLE
Automated build environment going out of support

### ISSUE THREAD

**Issue by spamatica** [muse-sequencer/muse#1096]
The automated builds for the compatibility build uses Ubuntu 18.04 which is out of support and will according to github be removed as a build environment from 2023-01-01.
We should update all build environments.

Was gonna give it a try to fix myself. Adding ticket here for visibility.

**Comment by terminator356** [comment]
G'day! I was visiting this problem to try and fix it.
Already, the 'compatibility build' fails, and the 'release build' may soon fail.

I could not tackle the problem at the moment since I don't have Ubuntu installed here,
 and I am not sure what packages need to change and so on.

Great that you'll take a look at it.

One thing that worried me was another big elephant in the room: Qt6
Soon all the distros might change over.
At some point we should sit down and try upgrading to see what happens,
 much like Qt4 - Qt5, I hope it will be relatively straight forward.

Nice to see ya again.
T.

**Comment by spamatica** [comment]
Yeah I know I have been very invisible... trying to get back in the game for a bit again. I would like to!

Yeah, I see. I thought the AppImage was 20.04 but it is actually 18.04 also so it needs to be updated asap too.
As I read the notice, 18.04 should function until end of year but will be down prioritized. So my guess is that the failed builds have been stopped by the build system due to high load (I didn't look at the logs though).

**Comment by spamatica** [comment]
The builds are now updated to use Ubuntu 22.04 image. Hopefully there aren't any immediate drawbacks with using such a recent distro as base. I've briefly tested the master image but since I'm on the same distro it might not mean much.
I tried with the 20.04 image but could not get the actual image working.

For future reference when we need to do this again.
Someone made an offline tool that emulates the behaviour of the github actions service. 
https://github.com/nektos/act
This made the process doable.  Doing it directly with github necessitates that you commit a change each time you need to test. Still takes quite a while as it requires docker and refreshing an image each time you test, but some of it I would just not have figured out without the offline tool and being able to peek inside the image.

**Comment by terminator356** [comment]
Yay. Thanks a lot! Great link.
It's unlikely that I will ever test an AppImage here,
 so it's over to the users now...

**Comment by revolverve** [comment]
Oh , i see , new build seems to work on 22.04 but not on previous version..or void linux
but old builds works on all of them .
Weird fact ,even extracted the latest wont load on 21.10 or 21.04 !!
Was that expected?
 note that i am not a user of ubuntu or void directly...only distros base on them. 
Was it wartime between pronobo/rh/flatpack/github?...636 comments on the topic..
Dont fight with appimage , its nice to have one for 22.04 and the next.
 But if curious....in impish;/root/MusE-master-x86_64.AppImage.extracted/usr/bin/muse4: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.35' not found (required by /root/MusE-master-x86_64.AppImage.extracted/usr/bin/../lib/libpython3.10.so.1.0)
/root/MusE-master-x86_64.AppImage.extracted/usr/bin/muse4: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.35' not found (required by /root/MusE-master-x86_64.AppImage.extracted/usr/bin/../lib/libQt5Core.so.5)

**Comment by spamatica** [comment]
> GLIBC_2.35' not found
Now that I see this error I realize this was what I was fearing. I think we may have had similar errors in the past.

Never heard about pronobo or rh. I can't vouch for exactly why AppImage was selected but it seems a solid choice once you get the dependencies as good as possible.

AppImage does suggest you start from the oldest supported image (to not have too recent dependencies)  so I tried again to use the 20.04, though I could not use it in my local testing. Seems like it did work on the build server so here's hoping the new AppImage is better supported.

Also, I realized lrdf was not installed so the new image has support for this too.
Lash support is disabled as I think the package has been removed.
